clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/SecurityBuilder.java

```diff
@@ -0,0 +1,84 @@
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.as.clustering.infinispan.subsystem.remote;
+
+import static org.jboss.as.clustering.infinispan.subsystem.remote.SecurityResourceDefinition.Attribute.SSL_CONTEXT;
+
+import javax.net.ssl.SSLContext;
+
+import org.infinispan.client.hotrod.configuration.ConfigurationBuilder;
+import org.infinispan.client.hotrod.configuration.SecurityConfiguration;
+import org.infinispan.client.hotrod.configuration.SecurityConfigurationBuilder;
+import org.jboss.as.clustering.controller.CommonUnaryRequirement;
+import org.jboss.as.clustering.controller.ResourceServiceBuilder;
+import org.jboss.as.clustering.infinispan.subsystem.ComponentBuilder;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.PathAddress;
+import org.jboss.dmr.ModelNode;
+import org.jboss.msc.service.ServiceBuilder;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceTarget;
+import org.jboss.msc.service.ValueService;
+import org.wildfly.clustering.service.Builder;
+import org.wildfly.clustering.service.InjectedValueDependency;
+import org.wildfly.clustering.service.ValueDependency;
+
+/**
+ * @author Radoslav Husar
+ */
+public class SecurityBuilder extends ComponentBuilder<SecurityConfiguration> implements ResourceServiceBuilder<SecurityConfiguration> {
+
+    private volatile ValueDependency<SSLContext> sslContextDependency;
+
+    SecurityBuilder(PathAddress remoteCacheContainer) {
+        super(RemoteCacheContainerComponent.SECURITY, remoteCacheContainer);
+    }
+
+    @Override
+    public Builder<SecurityConfiguration> configure(OperationContext context, ModelNode model) throws OperationFailedException {
+        ModelNode sslContextRef = SSL_CONTEXT.resolveModelAttribute(context, model);
+        if (sslContextRef.isDefined()) {
+            this.sslContextDependency = new InjectedValueDependency<>(CommonUnaryRequirement.SSL_CONTEXT.getServiceName(context, sslContextRef.asString()), SSLContext.class);
+        }
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder<SecurityConfiguration> build(ServiceTarget target) {
+        ServiceBuilder<SecurityConfiguration> builder = target.addService(this.getServiceName(), new ValueService<>(this)).setInitialMode(ServiceController.Mode.ON_DEMAND);
+        if (sslContextDependency != null) {
```

Comment:
return (this.sslContextDependency != null) ? this.sslContextDependency.register(builder) : builder;
:)

Reply:
LOL... ok.